src/main/java/pl/wrapper/parking/result/FailureResult.java

```diff
@@ -3,6 +3,7 @@
 
 import lombok.SneakyThrows;
 
+
 public record FailureResult<T,D extends Exception>(D error) implements Result<T> {
```

Comment:
if getValue() gets called on Failure, an exception is thrown. The point of the result pattern is to avoid throwing exceptions at all. The current implementation is simply changing the moment when they are thrown

Reply:
See my other comment on the Result interface for a general idea on how to avoid the exceptions being thrown